### ISSUE TITLE
:FZF works but not :Buffer, :Files, :History

### ISSUE THREAD

**Issue by the42** [junegunn/fzf.vim#1153]
- [x] I have fzf 0.23.0 or above
I have a system-wide fzf which is fzf 0.22. I have https://github.com/junegunn/fzf and did and ./install which fetched fzf 0.24
- [x] I have read through https://github.com/junegunn/fzf.vim/blob/master/README.md
- [x] I have read through https://github.com/junegunn/fzf/blob/master/README-VIM.md
- [x] I have read through the manual page of fzf (`man fzf`)
- [x] I have searched through the existing issues

I am using neovim head as of NVIM v0.5.0-782-gb6897ebc0

FZF was working for me without issues. Either a neovim update or a plugin-update caused fzf to stop working - but in a strange way.

When I run `:FZF` it works --> this is similar to https://github.com/junegunn/fzf.vim/issues/1102#issuecomment-716921535
When I run `:Files` or `:Buffer` the error message is:

```
Error detected while processing function fzf#vim#files[15]..<SNR>53_fzf[1]..<SNR>53_check_requirements:
line    9:
E605: Exception not caught: fzf#exec function not found. You need to upgrade Vim
 plugin from the main fzf repository ('junegunn/fzf')
```

When I run `:History` the error message is:

```
Error detected while processing function <SNR>35_history[7]..fzf#vim#history[1].
.<SNR>53_fzf[1]..<SNR>53_check_requirements:
line    9:
E605: Exception not caught: fzf#exec function not found. You need to upgrade Vim
 plugin from the main fzf repository ('junegunn/fzf')
Error detected while processing function <SNR>35_history:
line    7:
E171: Missing :endif
```

I absolutely updated to the latest fzf and fzf.vim

**Comment by junegunn** [comment]
> I absolutely updated to the latest fzf

Version?

**Comment by the42** [comment]
.config/nvim/pack/minpac/start/fzf/bin/fzf --version is 0.24

**Comment by junegunn** [comment]
Does `:echo fzf#exec()` work? The error message states that it's not available.

And if not, check if your Vim is loading the right file by examining `:scriptnames`

**Comment by the42** [comment]
`echo fzf#exec()` does not work and indeed `:scriptnames` returns among many other loaded scripts `/usr/share/vim/vimfiles/plugin/fzf.vim` before the local package is loaded. I guess the system-wide plugin is shadowing the later loaded fzf-package.

It is owned in Arch/Manjaro by the package `community/fzf 0.22.0-1` which in turn is required by another system package.

How can my local package take precedence over the system plugin?

**Comment by junegunn** [comment]
Hmm, I have no experience with minpac or package feature of Vim 8, so I have no idea.

**Comment by the42** [comment]
I solved it by adding `packadd fzf` right after package configuration of minpac in `init.vim`. This causes the package to load immediately instead of lazy loading upon request. By doing so the local package is sourced before the system package.

**Comment by aug0stus** [comment]
I had a similar issue, but I'm on Windows 10. 

01f7146cf6e903e31cd900944393bec86c6da4a6 introduced the function `s:check_requirements()` which uses `systemlist(...)` to get the version of fzf. Using procmon.exe, I identified that `systemlist(...)` somewhat uses the `shell` and redirects the output to a temporary file. My `shell` option is set to powershell.exe, which, by default, redirects (`>` & `>>` ) in UTF-16, and that doesn't seem to get parsed too well by vim after.

I added [$PSDefaultParameterValues['Out-File:Encoding'] = 'utf8'](https://stackoverflow.com/questions/40098771/changing-powershells-default-output-encoding-to-utf-8) in my `$PROFILE` and it's alright now :).

**Comment by bit101** [comment]
Had the exact same issue. After adding the `packadd fzf` line, I got past the earlier issues, but then got the error saying I had `fzf` version 2.2 and I needed 2.3 or higher. Upgraded to 2.4 and all was fine. I was able to remove the `packadd fzf` line and go back to my original config and all is good again.

**Comment by network-shark** [comment]
Just for help . My problem was that I updated the vim plugin , but had two fzf installs on my computer . From brew and from git clone . My vimrc pointed to and old version fzf in ~.fzf . Updated it and everything is working again !

**Comment by rrybarczyk** [comment]
Had the same problem as @network-shark and their fix worked for me on osx:
1. `% brew uninstall fzf`
2. `% rm ~/.fzf`
3. In my `init.vim`config, I removed all references to fzf, the did a `PlugClean` to fully uninstall.
4. I readded all my `fzf` related code and did a `PlugInstall`.
5. `fzf` is reinstalled and a new `~/.fzf` directory is automatically added.


For reference, my `fzf` related config:
```
set rtp+=~/.fzf
Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --all' }
Plug 'junegunn/fzf.vim'
let g:fzf_action = {
  \ 'ctrl-t': 'tab split',
  \ 'ctrl-s': 'split',
  \ 'ctrl-v': 'vsplit'
  \}

let $FZF_DEFAULT_COMMAND = 'rg --files'
```
